fhir-path/src/main/java/com/ibm/fhir/path/patch/FHIRPathPatch.java

```diff
@@ -17,33 +16,189 @@
 import com.ibm.fhir.model.resource.Parameters;
 import com.ibm.fhir.model.resource.Parameters.Parameter;
 import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.type.Code;
+import com.ibm.fhir.model.type.Element;
 
 public class FHIRPathPatch implements FHIRPatch {
-    private List<FHIRPathPatchOperation> operations;
+    private final List<FHIRPathPatchOperation> operations;
 
+    private FHIRPathPatch(Builder builder) {
+        this.operations = Collections.unmodifiableList(builder.operations);
+    }
+    
+    @Override
+    public <T extends Resource> T apply(T resource) throws FHIRPatchException {
+        for (FHIRPathPatchOperation fhirPathPatchOperation : operations) {
+            resource = fhirPathPatchOperation.apply(resource);
+        }
+        return resource;
+    }
+    
     /**
+     * Convert the FHIRPathPatch to a FHIR Parameters resource
+     */
+    public Parameters toParameters() {
+        Parameters.Builder builder = Parameters.builder();
+        for (FHIRPathPatchOperation operation : operations) {
+            builder.parameter(operation.toParameter());
+        }
+        return builder.build();
+    }
+    
+    public Builder toBuilder() {
+        return new Builder().from(this);
+    }
+    
+    public static Builder builder() {
+        return new Builder();
+    }
+    
+    public static class Builder { 
+        private List<FHIRPathPatchOperation> operations = new ArrayList<>(3);
+        
+        private Builder() {
+            // hidden constructor
+        }
+    
+        public Builder add(String path, String elementName, Element element) {
+            operations.add(new FHIRPathPatchAdd(path, elementName, element));
+            return this;
+        }
+        
+        public Builder delete(String path) {
+            operations.add(new FHIRPathPatchDelete(path));
+            return this;
+        }
+        
+        public Builder insert(String path, Element element, Integer index) {
+            operations.add(new FHIRPathPatchInsert(path, element, index));
+            return this;
+        }
+        
+        public Builder move(String path, Integer source, Integer destination) {
+            operations.add(new FHIRPathPatchMove(path, source, destination));
+            return this;
+        }
+        
+        public Builder replace(String path, Element element) {
+            operations.add(new FHIRPathPatchReplace(path, element));
+            return this;
+        }
+        
+        /**
+         * Build the {@link FHIRPathPatch}
+         * 
+         * @return
+         *     An immutable object of type {@link FHIRPathPatch}
+         */
+        public FHIRPathPatch build() {
+            return new FHIRPathPatch(this);
+        }
+
+        protected Builder from(FHIRPathPatch patch) {
```

Comment:
Is the intention of making this an instance method to support inter-mixing other patches with manually added operations like this?

```java
     FHIRPathPatch.builder()
          .add(...)
          .move(...)
          .from(someExistingPatch)
          .from(someOtherPatch)
          .build();
```

Or were you only thinking of the simple case of just wanting to mutate a single existing immutable `FHIRPathPatch`?

If you want to allow/support the former, then you should make this method public so anyone can do it. If not, then I'd suggest changing this to an additional private ctor which can only be used by the `toBuilder()` method of the outer class.

Reply:
> s the intention of making this an instance method to support inter-mixing other patches with manually added operations like this?

No, although I did consider something like that for supporting combining of disparate patches into a single patch.  I don't think it would be too hard to add, but since presently it just naively applies them one-at-a-time, I don't think it makes much difference whether you combine into a single patch or just execute them one-after-another.

> Or were you only thinking of the simple case of just wanting to mutate a single existing immutable FHIRPathPatch?

this

I just followed the precedent set in our generated model.  For example, see
https://github.com/IBM/FHIR/blob/master/fhir-model/src/main/java/com/ibm/fhir/model/resource/Account.java#L1381

I think an overloaded [protected] constructor is another option, but I think it just comes down to preference.  @JohnTimm any thoughts on this?

Reply:
Looking at it some more, I guess it wouldn't hurt to just flip this to public...then it could serve both use cases.  I can't think of a downside...

Reply:
done